src/system.cpp

```diff
@@ -788,6 +788,13 @@ void System::network_init(bool refresh) {
     //  ETH_CLOCK_GPIO17_OUT = 3  RMII clock output from GPIO17, for 50hz inverted clock
     auto clock_mode = (eth_clock_mode_t)eth_clock_mode_;
 
+    // reset power and add a delay as ETH doesn't not always start up correctly after a warm boot
+    // TODO still experimental
+    pinMode(eth_power_, OUTPUT);
+    digitalWrite(eth_power_, LOW);
+    delay(1000);
+    digitalWrite(eth_power_, HIGH);
+
 #if ESP_IDF_VERSION_MAJOR < 5
```

Comment:
Does this work? 
The comment says that eth_power can also be -1, this should be checked

Reply:
don't know! I must have done 30 uploads and hit the restart button at least 50 times and I can't reproduce it, even without this code.